fast/stages/00-bootstrap/variables.tf

```diff
@@ -28,6 +28,46 @@ variable "bootstrap_user" {
   default     = null
 }
 
+variable "cicd_config" {
+  description = "CI/CD configuration. Top-level providers can be created for subsequent stages. Set to null to disable, or set individual repositories to null if not needed."
+  type = object({
+    providers = list(string)
+    repositories = object({
+      bootstrap = object({
+        branch   = string
+        name     = string
+        provider = string
+      })
+      resman = object({
+        branch   = string
+        name     = string
+        provider = string
+      })
+    })
+  })
+  default = null
+  # validate top-level providers
+  validation {
+    condition = var.cicd_config == null ? true : alltrue([
+      for p in coalesce(var.cicd_config.providers, []) :
+      contains(["GITHUB", "GITLAB"], p)
+    ])
+    error_message = "Supported CI/CD providers: 'GITHUB', 'GITLAB'."
+  }
+  # validate repositories
+  validation {
+    condition = var.cicd_config == null ? true : alltrue([
+      for k, v in coalesce(var.cicd_config.repositories, {}) :
+      v == null || (
+        try(v.name, null) != null
+        &&
+        contains(["GITHUB", "GITLAB"], try(v.provider, "GITHUB"))
+      )
+    ])
+    error_message = "Non-null repositories need a valid name and provider."
+  }
+}
+
```

Comment:
Untested, but I think both conditions can be simplified a bit:
```hcl
  # validate top-level providers
  validation {
    condition = var.cicd_config == null || alltrue([
      for p in coalesce(var.cicd_config.providers, []) :
      contains(["GITHUB", "GITLAB"], p)
    ])
    error_message = "Supported CI/CD providers: 'GITHUB', 'GITLAB'."
  }
  # validate repositories
  validation {
    condition = alltrue([
      for _, config in try(var.cicd_config.repositories, {}) :
      try(config.name, null) != null &&
      contains(["GITHUB", "GITLAB"], try(config.provider, "GITHUB"))
    ])
    error_message = "Non-null repositories need a valid name and provider."
  }
```

Reply:
I amn. The `try(var.cicd_config.repositories, {})` does. If `var.cicd_config == null` the try returns `{}` and the whole expression is true